schema.ts

```diff
@@ -79,24 +79,46 @@ export interface BooleanSchema extends BaseSchema {
   type: ValueType.Boolean;
 }
 
-export interface NumberSchema extends BaseSchema {
+export type NumberSchema = CurrencySchema | BaseNumberSchema |
+  SliderSchema | ScaleSchema | NumericSchema | NumericDateSchema | NumericTimeSchema | NumericDateTimeSchema;
+export interface BaseNumberSchema extends BaseSchema {
   type: ValueType.Number;
-  codaType?: NumberHintTypes;
 }
 
-export interface NumericSchema extends NumberSchema {
+export interface NumericSchema extends BaseNumberSchema {
   codaType?: ValueHintType.Percent; // Can also be undefined if it's a vanilla number
   precision?: number;
   useThousandsSeparator?: boolean;
 }
 
+export interface NumericDateSchema extends BaseNumberSchema {
```

Comment:
We probably want to eliminate BaseDateSchema then and rename DateSchema-->StringDataSchema perhaps? We have the same issues with those 3 string-based schemas.